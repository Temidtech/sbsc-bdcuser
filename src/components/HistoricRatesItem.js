import React, { Component, Fragment } from 'react';
import { View, Text, Image } from 'react-native';
import moment from 'moment';
import { HistoricRatesStyles as styles } from '../styles';
import { getImageFromCurrency, trimRates } from '../dataApi';

export default class HistoricRatesItem extends Component {
  state = {
    ratesPerDate: this.props.ratesPerDate,
    date: this.props.date
  }

  static getDerivedStateFromProps(nextProps, prevState) {
    if (nextProps.ratesPerDate !== prevState.ratesPerDate) {
      return {
        ratesPerDate: nextProps.ratesPerDate,
      };
    }
    return null;
  }

  render() {
    const { ratesPerDate, date } = this.state;
    const sizedRates = trimRates(ratesPerDate);
    const formattedDate = moment(date).format('ddd, MMM D');

    return (
      <Fragment>
        <View style={styles.bodyRowView}>
          <View style={styles.dateColumn}>
            <Text style={styles.dateText}>{formattedDate}</Text>
          </View>
          <Fragment>
            <View style={styles.currDataView}>
              {sizedRates.map((rate, index) => (
                !!rate && <View style={styles.currItemView} key={index}>
                  <Image source={getImageFromCurrency(rate.currency)} style={styles.currImage}/>
                  <Text style={styles.ratesText}>{rate.currency}</Text>
                </View>
              ))}
            </View>
            <View style={styles.ratesItemView}>
              {sizedRates.map((rate, index) => (
                !!rate && <Text key={index} style={styles.ratesText}>{rate.buyRate}</Text>
              ))}
            </View>
            <View style={styles.ratesItemView}>
              {sizedRates.map((rate, index) => (
                !!rate && <Text key={index} style={styles.ratesText}>{rate.sellRate}</Text>
              ))}
            </View>
          </Fragment>
        </View>
      </Fragment>
    );
  }
}
