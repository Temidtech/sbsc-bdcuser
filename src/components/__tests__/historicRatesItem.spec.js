import React from 'react';
import renderer from 'react-test-renderer';
import HistoricRatesItem from '../HistoricRatesItem';

describe('<HistoricRatesItem />', () => {
  const testProps = {
    ratesPerDate: [
      {
        date: '2018-06-07',
        currency: 'EUR',
        buyRate: 415,
        sellRate: 418,
        periodOfTheDay: 'AFTERNOON',
      },
      { date: '2018-06-07',
        currency: 'USD',
        buyRate: 330,
        sellRate: 347,
        periodOfTheDay: 'MORNING',
      }
    ],
    date: '2018-06-07'
  };

  it('renders the HistoricRatesItem component correctly', () => {
    const tree = renderer.create(
      <HistoricRatesItem {...testProps} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
