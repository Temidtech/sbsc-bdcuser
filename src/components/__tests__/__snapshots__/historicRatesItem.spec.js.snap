// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HistoricRatesItem /> renders the HistoricRatesItem component correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "width": "30%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "yellow",
          "fontFamily": "OpenSans-Italic",
        }
      }
    >
      Thu, Jun 7
    </Text>
  </View>
  <View
    style={
      Object {
        "marginBottom": 10,
        "width": "20%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/img/usd.png",
          }
        }
        style={
          Object {
            "height": 10,
            "marginRight": 5,
            "width": 20,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontFamily": "OpenSans-Regular",
            "marginBottom": 5,
          }
        }
      >
        USD
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/img/eur.png",
          }
        }
        style={
          Object {
            "height": 10,
            "marginRight": 5,
            "width": 20,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontFamily": "OpenSans-Regular",
            "marginBottom": 5,
          }
        }
      >
        EUR
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 10,
        "width": "25%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "OpenSans-Regular",
          "marginBottom": 5,
        }
      }
    >
      330
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "OpenSans-Regular",
          "marginBottom": 5,
        }
      }
    >
      415
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 10,
        "width": "25%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "OpenSans-Regular",
          "marginBottom": 5,
        }
      }
    >
      347
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "OpenSans-Regular",
          "marginBottom": 5,
        }
      }
    >
      418
    </Text>
  </View>
</View>
`;
